exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -317,3 +325,255 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 
 	return ctrl.Result{}, nil
 }
+
+// reconcileMachinePoolMachines reconciles MachinePool Machines associated with a MachinePool.
+func (r *MachinePoolReconciler) reconcileMachinePoolMachines(ctx context.Context, mp *expv1.MachinePool, infraMachinePool *unstructured.Unstructured) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	var noKind bool
+	var noSelector bool
+
+	var infraMachineKind string
+	if err := util.UnstructuredUnmarshalField(infraMachinePool, &infraMachineKind, "status", "infrastructureMachineKind"); err != nil {
+		if errors.Is(err, util.ErrUnstructuredFieldNotFound) {
+			noKind = true
+		} else {
+			return errors.Wrapf(err, "failed to retrieve infraMachineKind from infrastructure provider for MachinePool %q in namespace %q", mp.Name, mp.Namespace)
+		}
+	}
+
+	var infraMachineSelector metav1.LabelSelector
+	if err := util.UnstructuredUnmarshalField(infraMachinePool, &infraMachineSelector, "status", "infrastructureMachineSelector"); err != nil {
+		if errors.Is(err, util.ErrUnstructuredFieldNotFound) {
+			noSelector = true
+		} else {
+			return errors.Wrapf(err, "failed to retrieve infraMachineSelector from infrastructure provider for MachinePool %q in namespace %q", mp.Name, mp.Namespace)
+		}
+	}
+
+	// If one missing but not both, return error
+	if noKind && noSelector {
+		// If proper MachinePool Machines aren't supported by the InfraMachinePool, just create Machines to match the replica count.
+		log.Info("No infraMachineKind or infraMachineSelector found, reconciling fallback machines")
+		return r.reconcileFallbackMachines(ctx, mp)
+	} else if noKind != noSelector {
+		log.Info("Only one of infraMachineKind or infraMachineSelector found, this is unexpected")
+		return errors.Errorf("only one of infraMachineKind or infraMachineSelector found, both must be present")
+	}
+
+	log.Info("Reconciling MachinePool Machines", "infrastructureMachineKind", infraMachineKind, "infrastructureMachineSelector", infraMachineSelector)
```

Comment:
I think yes, because of the point I raised here: https://github.com/kubernetes-sigs/cluster-api/pull/7938#discussion_r1156230404

Right now, the Infra Machine might be deleted for several minutes before CAPI reacts.

Reply:
I think that's fair, I'll go ahead and add that